flow/cmd/validate_mirror.go

```diff
@@ -190,3 +196,93 @@ func (h *FlowRequestHandler) CheckIfMirrorNameExists(ctx context.Context, mirror
 
 	return nameExists.Bool, nil
 }
+
+func (h *FlowRequestHandler) ValidateTableAdditions(ctx context.Context, req *protos.ValidateTableAdditionsRequest) (*protos.ValidateCDCMirrorResponse, error) {
+	sourcePeer, err := connectors.LoadPeer(ctx, h.pool, req.SourcePeerName)
+	if err != nil {
+		slog.Error("table addition validation: failed to load source peer", slog.String("peer", req.SourcePeerName))
+		return nil, err
+	}
+
+	sourcePeerConfig := sourcePeer.GetPostgresConfig()
+	if sourcePeerConfig == nil {
+		slog.Error("table addition validation: source peer config is not postgres", slog.String("peer", req.SourcePeerName))
+		return nil, errors.New("source peer config is not postgres")
+	}
+
+	pgPeer, err := connpostgres.NewPostgresConnector(ctx, nil, sourcePeerConfig)
```

Comment:
We need to stop creating postgres connectors like this in endpoints

See #2525 where I put validation behind a connector interface